### PR TITLE
databases-week3

### PR DESCRIPTION
**Pull request by ReenuMary**

### CODE REVIEW

File: databases/week3/meal_sharing_queries.sql [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+DELETE FROM review WHERE id=8;
+
+#Get meals that has a price smaller than a specific price fx 90
+SELECT * FROM meal WHERE price<=200;
+#Get meals that still has available reservations
+SELECT reservation.meal_id,meal.title, meal.max_reservations ,sum( reservation.number_of_guests) 
+FROM reservation
+JOIN meal ON 
+reservation.meal_id = meal.id
+GROUP BY reservation.meal_id;
```
Comment (Elia-Git) [inline databases/week3/meal_sharing_queries.sql:70]: What are you aiming to get here..? I can't see which meal is fully or partially booked. Each meal has a limited nr. of spots to reserve &we need to know which meals are not fully booked/reserved..?
Try this instead & see how big a difference it makes in the result
```
SELECT meal.*, sum(reservation.number_of_guests) AS Total_reserved_yet
FROM meal
JOIN reservation ON meal.id = reservation.meal_id
GROUP BY reservation.meal_id
HAVING meal.max_reservations > Total_reserved_yet;
```

File: databases/week3/meal_sharing_queries.sql [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+SELECT * FROM meal WHERE title LIKE '%chicken%';
+#Get meals that has been created between two dates
+SELECT * FROM meal WHERE created_date BETWEEN '2022-08-01' AND '2022-08-30';
+#Get only specific number of meals fx return only 5 meals
+SELECT * FROM meal LIMIT 3;
+#Get the meals that have good reviews
+SELECT review.meal_id,meal.title,review.stars,COUNT(review.stars) AS 'No of reviews with this rating' FROM review 
+JOIN meal ON review.meal_id= meal.id
+WHERE review.stars>2
+GROUP BY review.stars, review.meal_id;
```
Comment (Elia-Git) [inline databases/week3/meal_sharing_queries.sql:82]: Here why do we need to count stars? We can specify when we call a meal got a good, bad or moderate review, then we can filter.
In this case, if I say reviews>10 -good, review between 5-9 moderate & reviews <5 could be bad reviewed
and we can try like below
```
SELECT meal.*,review.stars AS Stars_recieved FROM meal
JOIN review ON meal.id=review.Meal_Id
WHERE review.stars>9;
```

**Review by Elia-Git — CHANGES_REQUESTED** [review]
Few comments for a quick fix. Thanks! # is not a comment syntax in VS code, pls mark your comment lines with -- instead of #. Make sure to try your code is running in VS code before pushing

File: databases/week3/meal_sharing_create_db.sql [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  `created_date` DATE NULL,
+  PRIMARY KEY (`id`),
+  INDEX `fk_review_meal_id_idx` (`meal_id` ASC) VISIBLE,
+  CONSTRAINT `fk_review_meal_id`
+    FOREIGN KEY (`meal_id`)
+    REFERENCES `meal_sharing`.`meal` (`id`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE)
+ENGINE = InnoDB;
+
```
Comment (shravanipagilla) [inline databases/week3/meal_sharing_create_db.sql:75]: Hej @ReenuMary  ,can you upload the .png  file  :)  ..   if you can ...

File: databases/week3/meal_sharing_queries.sql [hunk 4]
```diff
@@ -0,0 +1,92 @@
+use meal_sharing;
+
+#Get all meals
+SELECT * FROM meal;
+#Get a meal with any id, fx 1
+SELECT * FROM meal WHERE id=2;
+#Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE meal SET title ='Vegetable Korma', 
+description ='This delectable Indian Vegetable Korma recipe is loaded with potatoes, tomatoes, carrots, peas, and green beans. Have plenty of naan to dip!'
```
Comment (shravanipagilla) [inline databases/week3/meal_sharing_queries.sql:9]: :) yummy food...

File: databases/week3/meal_sharing_queries.sql [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+'2022-09-29');
+#Get a review with any id, fx 1
+SELECT * FROM review WHERE id=2;
+#Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE review SET title='Bad'  WHERE id=2;
+#Delete a review with any id, fx 1
+DELETE FROM review WHERE id=8;
+
+#Get meals that has a price smaller than a specific price fx 90
+SELECT * FROM meal WHERE price<=200;
```
Comment (shravanipagilla) [inline databases/week3/meal_sharing_queries.sql:64]: 👍

File: databases/week3/meal_sharing_queries.sql (hunk 1 above)
Comment (shravanipagilla) [reply databases/week3/meal_sharing_queries.sql:70]: 👍

File: databases/week3/meal_sharing_queries.sql [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+GROUP BY reservation.meal_id;
+
+#Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT * FROM meal WHERE title LIKE '%chicken%';
+#Get meals that has been created between two dates
+SELECT * FROM meal WHERE created_date BETWEEN '2022-08-01' AND '2022-08-30';
+#Get only specific number of meals fx return only 5 meals
+SELECT * FROM meal LIMIT 3;
+#Get the meals that have good reviews
+SELECT review.meal_id,meal.title,review.stars,COUNT(review.stars) AS 'No of reviews with this rating' FROM review
```
Comment (shravanipagilla) [inline databases/week3/meal_sharing_queries.sql:79]: Can you write AS is little short form , it is just my suggestion . if you want .. :)

**Review by shravanipagilla — COMMENTED** [review]
👍  I hope you can understand my comments otherwise just  let me know @ReenuMary .. 
thank you .

**Pull request merged**